### PR TITLE
CPM-724: Add Product Uuids endpoints

### PR DESCRIPTION
**Pull request by pierallard**
TODO
- [x] Test Postman
- [x] Check to execute each endpoint with real examples

@meganelepalud-akeneo Should we invert Product [uuid] and Product [identifier] in the list to have uuid first ?

### CODE REVIEW

**Review by tseho — APPROVED** [review]

**Review by LevFlavien — COMMENTED** [review]
- Update JSON examples like [here](https://github.com/akeneo/pim-api-docs/blob/7b5fb15ca56906db896a281cd72b8795fa8170ff/content/concepts/products.md?plain=1#L29) (twice in this file)
- Each API reference route has its own anchor link (eg /api-reference.html#post_products_uuid), check for them in the doc 
 -  API error [messages](https://github.com/akeneo/pim-api-docs/blob/7b5fb15ca56906db896a281cd72b8795fa8170ff/content/rest-api/responses.md?plain=1#L219) should be updated. :warning: Maybe the PIM needs to be updated too

File: content/files/Akeneo PIM API.postman_collection.json [hunk 1]
```diff
@@ -3728,6 +3728,493 @@
...
... \"data\": \"black\"\n      }\n    ],\n    \"name\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"Akeneo T-Shirt black and purple with short sleeve\"\n      }\n    ],\n    \"secondary_color\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"purple\"\n      }\n    ],\n    \"tshirt_materials\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"cotton\"\n      }\n    ],\n    \"tshirt_style\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": [\n          \"crewneck\",\n          \"short_sleeve\"\n        ]\n      }\n    ],\n    \"price\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": [\n          {\n            \"amount\": 10,\n            \"currency\": \"EUR\"\n          },\n          {\n            \"amount\": 14,\n            \"currency\": \"USD\"\n          }\n        ]\n      }\n    ]\n  },\n  \"created\": \"2017-03-30T14:55:26+02:00\",\n  \"updated\": \"2017-05-04T23:56:09+02:00\",\n  \"associations\": {\n    \"SUBSTITUTION\": {\n      \"groups\": [\"tshirt\"],\n      \"products\": []\n    }\n  }\n}"
```
Comment (LevFlavien) [inline content/files/Akeneo PIM API.postman_collection.json:4031]: add `uuid` property?

Comment (LevFlavien) [reply content/files/Akeneo PIM API.postman_collection.json:4031]: same for PATCH product

Comment (tseho) [inline content/files/Akeneo PIM API.postman_collection.json:4031]:
```suggestion
              "raw": "{\n  \"family\": \"tshirts\",\n  \"groups\": [],\n  \"parent\": null,\n  \"categories\": [\n    \"tvs_projectors\"\n  ],\n  \"enabled\": true,\n  \"values\": {\n    \"sku\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"new_product\"\n      }\n    ],\n    \"clothing_size\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"s\"\n      }\n    ],\n    \"description\": [\n      {\n        \"locale\": \"en_US\",\n        \"scope\": \"mobile\",\n        \"data\": \"Akeneo T-Shirt 77777\"\n      },\n      {\n        \"locale\": \"en_US\",\n        \"scope\": \"print\",\n        \"data\": \"Akeneo T-Shirt with short sleeve\"\n      }\n    ],\n    \"main_color\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"black\"\n      }\n    ],\n    \"name\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"Akeneo T-Shirt black and purple with short sleeve\"\n      }\n    ],\n    \"secondary_color\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"purple\"\n      }\n    ],\n    \"tshirt_materials\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": \"cotton\"\n      }\n    ],\n    \"tshirt_style\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": [\n          \"crewneck\",\n          \"short_sleeve\"\n        ]\n      }\n    ],\n    \"price\": [\n      {\n        \"locale\": null,\n        \"scope\": null,\n        \"data\": [\n          {\n            \"amount\": 10,\n            \"currency\": \"EUR\"\n          },\n          {\n            \"amount\": 14,\n            \"currency\": \"USD\"\n          }\n        ]\n      }\n    ]\n  },\n  \"created\": \"2017-03-30T14:55:26+02:00\",\n  \"updated\": \"2017-05-04T23:56:09+02:00\",\n  \"associations\": {\n    \"SUBSTITUTION\": {\n      \"groups\": [\"promotion\"],\n      \"products\": []\n    }\n  }\n}"
```

unless I'm mistaken, the group "tshirt" doesn't exists in icecat, only "promotion".

Comment (pierallard) [reply content/files/Akeneo PIM API.postman_collection.json:4031]: uuid is not mandatory for this endpoint

**Comment by pierallard** [comment]
> * Update JSON examples like [here](https://github.com/akeneo/pim-api-docs/blob/7b5fb15ca56906db896a281cd72b8795fa8170ff/content/concepts/products.md?plain=1#L29) (twice in this file)
> * Each API reference route has its own anchor link (eg /api-reference.html#post_products_uuid), check for them in the doc
> * API error [messages](https://github.com/akeneo/pim-api-docs/blob/7b5fb15ca56906db896a281cd72b8795fa8170ff/content/rest-api/responses.md?plain=1#L219) should be updated. warning Maybe the PIM needs to be updated too

Done !

File: content/files/Akeneo PIM API.postman_collection.json (hunk 1 above)
Comment (LevFlavien) [reply content/files/Akeneo PIM API.postman_collection.json]: ok, and for unitary PATCH actually it's in the url.

File: content/swagger/resources/products_uuid/routes/products_uuid_uuid_draft.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+get:
+  summary: Get a draft
+  operationId: "get_draft_uuid__code_"
```
Comment (LevFlavien) [inline content/swagger/resources/products_uuid/routes/products_uuid_uuid_draft.yaml:3]:
```suggestion
  operationId: "get_draft_uuid__uuid_"
```

**Review by LevFlavien — COMMENTED** [review]
Still some API ref links to replace (unless it's not relevant with the content):
- `/api-reference.html#get_products` -> `/api-reference.html#get_products_uuid`
- `/api-reference.html#patch_products` -> `/api-reference.html#patch_products_uuid`
- etc

File: content/files/Akeneo PIM API.postman_collection.json [hunk 3]
```diff
@@ -3728,6 +3728,493 @@
...
+                "products-uuid",
+                "f62190e8-8314-4360-8ff8-7dbdb4af6312"
+              ]
+            },
+            "description": "Assuming that the given uuid is the uuid of an existing product"
+          },
+          "response": []
+        },
+        {
+          "name": "draft (2.x and EE only)",
```
Comment (LevFlavien) [inline content/files/Akeneo PIM API.postman_collection.json:4153]:
```suggestion
          "name": "draft (SaaS only)",
```

File: content/files/Akeneo PIM API.postman_collection.json [hunk 4]
```diff
@@ -3728,6 +3728,493 @@
...
+                "f62190e8-8314-4360-8ff8-7dbdb4af6312",
+                "draft"
+              ]
+            },
+            "description": "Assuming that there is already a draft for the given product. The draft was created by the same user that using this request.\nThe user has only an edition permission through categories on the given product."
+          },
+          "response": []
+        },
+        {
+          "name": "proposal (2.x and EE only)",
```
Comment (LevFlavien) [inline content/files/Akeneo PIM API.postman_collection.json:4181]:
```suggestion
          "name": "proposal (SaaS only)",
```

File: content/files/Akeneo PIM API.postman_collection.json [hunk 5]
```diff
@@ -3728,6 +3728,493 @@
         }
       ]
     },
+    {
+      "name": "Product (uuid)",
```
Comment (LevFlavien) [inline content/files/Akeneo PIM API.postman_collection.json:3732]: Add `(SaaS only)` for collection folder and each route

**Comment by meganelepalud-akeneo** [comment]
Should we invert Product [uuid] and Product [identifier] in the list to have uuid first ?

@pierallard don't bother. We'll work on a deprecation policy probably next year

**Comment by LevFlavien** [comment]
> don't bother.

@meganelepalud-akeneo just fyi it's not complex to do at all

**Review by LevFlavien — APPROVED** [review]

**Pull request merged**